master/pkg/schemas/expconf/hyperparameters_config.go

```diff
@@ -22,6 +47,8 @@ type HyperparameterV0 struct {
 	RawDoubleHyperparameter      *DoubleHyperparameterV0      `union:"type,double" json:"-"`
 	RawLogHyperparameter         *LogHyperparameterV0         `union:"type,log" json:"-"`
 	RawCategoricalHyperparameter *CategoricalHyperparameterV0 `union:"type,categorical" json:"-"`
+	// This option does not actually go through union marshaling and unmarshaling logic.
```

Comment:
Don't be afraid to leave more detail!

If I walked up to this code without any background understanding, I would not know right away "why did they bring up this detail at all?"

Can you go into more detail about "we're using the `union` tag due to XYZ, even though we specifically avoid the `union` behavior for marshaling/unmarshaling this field"